src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -1282,11 +1285,7 @@ bool StatefulWriter::matched_reader_remove(
 
         if (nullptr != listener_)
         {
-            // call the listener without locks taken
-            guard_locator_selector_async.unlock();
-            guard_locator_selector_general.unlock();
-            lock.unlock();
```

Comment:
The current changes make the listener be called with the writer's mutex taken. I don't know whether we want that behavior change.

Reply:
Nice catch. No, we should keep the former behavior and call it lock free